src/outlet/index.ts

```diff
@@ -167,21 +171,28 @@ export class Outlet {
 
 		if ((!window.parent && !window.opener) || !document.referrer)
 			return;
-
+		
 		const origin = new URL(document.referrer).origin;
-
+		
 		if (origin === document.location.origin)
-			return;
-
+		return;
+		
 		let accessWhitelist = JSON.parse(localStorage.getItem("tn-whitelist")) ?? {};
-		const storageRequestNeeded = window.parent && (document.hasStorageAccess && !(await document.hasStorageAccess()));
+		// window.parent always defined, so no need to check
+		if (!document.hasStorageAccess) {
```

Comment:
@oleggrib Even if storage access isn't required, we still need to add the site to the whitelist. This will allow old browsers to bypass the permission dialog.

Reply:
old browsers dont have this method _document.hasStorageAccess_ , so whole flow will be ignored. is it ok?

Reply:
No I think we always want to show permission dialog. It's better for security & user privacy. We don't want off-chain tokens to be used as a tracking cookie.

Reply:
@oleggrib In the interest of keeping momentum have committed proposed changes to this branch. If you are okay with this let's get this merged, otherwise let's have a discussion and we can revert if needed.